modules/src/ics02_client/events.rs

```diff
@@ -7,50 +7,50 @@ use anomaly::BoxError;
 
 use crate::ics02_client::height::Height;
 use serde_derive::{Deserialize, Serialize};
-use std::collections::HashSet;
 use std::convert::{TryFrom, TryInto};
 use tendermint::block;
 
 /// The content of the `type` field for the event that a chain produces upon executing the create client transaction.
-pub const CREATE_EVENT_TYPE: &str = "create_client";
-pub const UPDATE_EVENT_TYPE: &str = "update_client";
+const CREATE_EVENT_TYPE: &str = "create_client";
+const UPDATE_EVENT_TYPE: &str = "update_client";
 
 /// The content of the `key` field for the attribute containing the client identifier.
-pub const CLIENT_ID_ATTRIBUTE_KEY: &str = "client_id";
+const CLIENT_ID_ATTRIBUTE_KEY: &str = "client_id";
 
 /// The content of the `key` field for the attribute containing the client type.
-pub const CLIENT_TYPE_ATTRIBUTE_KEY: &str = "client_type";
+const CLIENT_TYPE_ATTRIBUTE_KEY: &str = "client_type";
 
 /// The content of the `key` field for the attribute containing the height.
-pub const CONSENSUS_HEIGHT_ATTRIBUTE_KEY: &str = "consensus_height";
-
-/// A list of all the event `type`s that this module is capable of parsing
-fn event_types() -> HashSet<String> {
-    vec![CREATE_EVENT_TYPE.to_string(), UPDATE_EVENT_TYPE.to_string()]
-        .into_iter()
-        .collect()
+const CONSENSUS_HEIGHT_ATTRIBUTE_KEY: &str = "consensus_height";
+
+pub fn try_from_tx(event: &tendermint::abci::Event) -> Option<IBCEvent> {
```

Comment:
This PR touches the translation from `tendermint::abci::Event` to `IBCEvent`  (here and in other modules) but didn't have to. This was done while generalizing the translation to not be specific to `tendermint::abci::Event`. This turned out to not be necessary after removing the `ibc::handler::Event`. Nevertheless,  in the process of generalising this, some improvements were made and we may want to keep them.